charts/csi-secret-provider-class/templates/deployment.yaml

```diff
@@ -0,0 +1,46 @@
+{{- if .Values.dummyDeployment.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  {{ template "common.metadata" $ }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "csi-secret-provider-class.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.dummyDeployment.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "csi-secret-provider-class.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: dummy
+          image: {{ .Values.dummyDeployment.image.repository }}:{{ .Values.dummyDeployment.image.tag }}
+          imagePullPolicy: {{ .Values.dummyDeployment.image.pullPolicy }}
+          {{- with .Values.dummyDeployment.resources }}
+          resources:
+            {{- toYaml .Values.dummyDeployment.resources | nindent 12 }}
+          {{- end }}
+      {{- with .Values.dummyDeployment.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.dummyDeployment.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.dummyDeployment.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      volumes:
+        - name: csi-secret-store-volume
```

Comment:
Is it sufficient to define the volume in `volumes` without defining a `volumeMount` in any container?

Reply:
yes.

Reply:
s/yes/maybe/ 🤨

Reply:
AH!

Reply:
But not by the "dummyDeployment" which is NOT using the Secret right away.

Reply:
we might want to investigate this further, i can't really explain what has been happeining yet but i can take another deep dive into the implementation to figure it out if we don't identify a pattern.